From: Cordell Bloor <cgmb@slerp.xyz>
Date: Fri, 15 Mar 2024 00:45:35 -0600
Subject: expand endian detection

The endianness detection logic in hsa/hsa.h fails on arm64 and ppc64el, which
leads to build failures in rccl on those platforms.

Forwarded: no
Bug-Debian: https://bugs.debian.org/1065695
---
 src/inc/hsa.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/inc/hsa.h b/src/inc/hsa.h
index a70fd0f..99c9f3d 100644
--- a/src/inc/hsa.h
+++ b/src/inc/hsa.h
@@ -79,9 +79,12 @@
 
 // Try to detect CPU endianness
 #if !defined(LITTLEENDIAN_CPU) && !defined(BIGENDIAN_CPU)
-#if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || \
+#if (defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__) || \
+    defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || \
     defined(_M_X64)
 #define LITTLEENDIAN_CPU
+#elif (defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+#define BIGENDIAN_CPU
 #endif
 #endif
 
